// @pulumi/azure-native/types/enums/networkanalytics/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedServiceIdentityType = exports.DefaultAction = exports.DataTypeState = exports.ControlState = exports.v20231115 = void 0;
// Export sub-modules:
const v20231115 = require("./v20231115");
exports.v20231115 = v20231115;
exports.ControlState = {
    /**
     * Field to enable a setting.
     */
    Enabled: "Enabled",
    /**
     * Field to disable a setting.
     */
    Disabled: "Disabled",
};
exports.DataTypeState = {
    /**
     * Field to specify stopped state.
     */
    Stopped: "Stopped",
    /**
     * Field to specify running state.
     */
    Running: "Running",
};
exports.DefaultAction = {
    /**
     * Represents allow action.
     */
    Allow: "Allow",
    /**
     * Represents deny action.
     */
    Deny: "Deny",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
