// @pulumi/azure-native/types/enums/authorization/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SelectorKind = exports.ResourceIdentityType = exports.PublicNetworkAccessOptions = exports.PrincipalType = exports.PolicyType = exports.ParameterType = exports.OverrideKind = exports.LockLevel = exports.ExemptionCategory = exports.EnforcementMode = exports.DefaultDecisionType = exports.AssignmentScopeValidation = exports.AccessReviewResult = exports.AccessReviewRecurrenceRangeType = exports.AccessReviewRecurrencePatternType = exports.v20250101 = exports.v20240501 = exports.v20240401 = exports.v20230401 = exports.v20220701preview = exports.v20220601 = exports.v20220401 = exports.v20211201preview = exports.v20210601 = exports.v20200501 = exports.v20200401preview = exports.v20200301preview = exports.v20200301 = exports.v20190601 = exports.v20180501 = void 0;
// Export sub-modules:
const v20180501 = require("./v20180501");
exports.v20180501 = v20180501;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20220401 = require("./v20220401");
exports.v20220401 = v20220401;
const v20220601 = require("./v20220601");
exports.v20220601 = v20220601;
const v20220701preview = require("./v20220701preview");
exports.v20220701preview = v20220701preview;
const v20230401 = require("./v20230401");
exports.v20230401 = v20230401;
const v20240401 = require("./v20240401");
exports.v20240401 = v20240401;
const v20240501 = require("./v20240501");
exports.v20240501 = v20240501;
const v20250101 = require("./v20250101");
exports.v20250101 = v20250101;
exports.AccessReviewRecurrencePatternType = {
    Weekly: "weekly",
    AbsoluteMonthly: "absoluteMonthly",
};
exports.AccessReviewRecurrenceRangeType = {
    EndDate: "endDate",
    NoEnd: "noEnd",
    Numbered: "numbered",
};
exports.AccessReviewResult = {
    Approve: "Approve",
    Deny: "Deny",
    NotReviewed: "NotReviewed",
    DontKnow: "DontKnow",
    NotNotified: "NotNotified",
};
exports.AssignmentScopeValidation = {
    /**
     * This option will validate the exemption is at or under the assignment scope.
     */
    Default: "Default",
    /**
     * This option will bypass the validation the exemption scope is at or under the policy assignment scope.
     */
    DoNotValidate: "DoNotValidate",
};
exports.DefaultDecisionType = {
    Approve: "Approve",
    Deny: "Deny",
    Recommendation: "Recommendation",
};
exports.EnforcementMode = {
    /**
     * The policy effect is enforced during resource creation or update.
     */
    Default: "Default",
    /**
     * The policy effect is not enforced during resource creation or update.
     */
    DoNotEnforce: "DoNotEnforce",
};
exports.ExemptionCategory = {
    /**
     * This category of exemptions usually means the scope is not applicable for the policy.
     */
    Waiver: "Waiver",
    /**
     * This category of exemptions usually means the mitigation actions have been applied to the scope.
     */
    Mitigated: "Mitigated",
};
exports.LockLevel = {
    NotSpecified: "NotSpecified",
    CanNotDelete: "CanNotDelete",
    ReadOnly: "ReadOnly",
};
exports.OverrideKind = {
    /**
     * It will override the policy effect type.
     */
    PolicyEffect: "policyEffect",
};
exports.ParameterType = {
    String: "String",
    Array: "Array",
    Object: "Object",
    Boolean: "Boolean",
    Integer: "Integer",
    Float: "Float",
    DateTime: "DateTime",
};
exports.PolicyType = {
    NotSpecified: "NotSpecified",
    BuiltIn: "BuiltIn",
    Custom: "Custom",
    Static: "Static",
};
exports.PrincipalType = {
    User: "User",
    Group: "Group",
    ServicePrincipal: "ServicePrincipal",
    ForeignGroup: "ForeignGroup",
    Device: "Device",
};
exports.PublicNetworkAccessOptions = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    /**
     * Indicates that a system assigned identity is associated with the resource.
     */
    SystemAssigned: "SystemAssigned",
    /**
     * Indicates that a system assigned identity is associated with the resource.
     */
    UserAssigned: "UserAssigned",
    /**
     * Indicates that no identity is associated with the resource or that the existing identity should be removed.
     */
    None: "None",
};
exports.SelectorKind = {
    /**
     * The selector kind to filter policies by the resource location.
     */
    ResourceLocation: "resourceLocation",
    /**
     * The selector kind to filter policies by the resource type.
     */
    ResourceType: "resourceType",
    /**
     * The selector kind to filter policies by the resource without location.
     */
    ResourceWithoutLocation: "resourceWithoutLocation",
    /**
     * The selector kind to filter policies by the policy definition reference ID.
     */
    PolicyDefinitionReferenceId: "policyDefinitionReferenceId",
};
