// @pulumi/azure-native/networkanalytics/v20231115/dataType.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The data type resource.
 */
class DataType extends pulumi.CustomResource {
    /**
     * Get an existing DataType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DataType(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataType.__pulumiType;
    }
    /**
     * Create a DataType resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataProductName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataProductName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["dataProductName"] = args ? args.dataProductName : undefined;
            resourceInputs["dataTypeName"] = args ? args.dataTypeName : undefined;
            resourceInputs["databaseCacheRetention"] = args ? args.databaseCacheRetention : undefined;
            resourceInputs["databaseRetention"] = args ? args.databaseRetention : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["state"] = args ? args.state : undefined;
            resourceInputs["storageOutputRetention"] = args ? args.storageOutputRetention : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["stateReason"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["visualizationUrl"] = undefined /*out*/;
        }
        else {
            resourceInputs["databaseCacheRetention"] = undefined /*out*/;
            resourceInputs["databaseRetention"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["stateReason"] = undefined /*out*/;
            resourceInputs["storageOutputRetention"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["visualizationUrl"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:networkanalytics:DataType" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DataType.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DataType = DataType;
/** @internal */
DataType.__pulumiType = 'azure-native:networkanalytics/v20231115:DataType';
